client/src/components/Group.jsx

```diff
@@ -1,70 +1,128 @@
+    AppActions.loadGroups();
   }
 
-  render () {
-    return (
-      <div className="container">
-      <h2>Activate Modal with JavaScript</h2>
- 
-      <button type="button"
-       className="btn btn-info default"
-       id="myBtn">
-       Open Modal
-       </button>
+  // loadGroupMessages(groupName) {
 
-      <div className="modal fade"
-       id="myModal"
-       role="dialog">
-        <div className="modal-dialog">
-      
-          <div className="modal-content">
-            <div className="modal-header">
-              <button type="button" className="close" data-dismiss="modal">&times;</button>
-              <h4 className="modal-title">Modal Header</h4>
-            </div>
-            <div className="modal-body">
-              <p>Some text in the modal.</p>
-            </div>
-            <div className="modal-footer">
-              <button type="button" className="btn btn-default" data-dismiss="modal">Close</button>
-            </div>
-          </div>
-      
-          </div>
+  // }
+
+  /**
+   *
+   * @returns
+   * @memberof Group
+   */
+  render() {
+    const waiting = 'Waiting';
+    return (
+         <div>
+           <form id='group-form'>
+              <h4><center> Group List</center><hr/></h4>
+              <button id = 'modal-button'type="button"
+               className="btn btn-success"
+               data-toggle="modal"
+               data-target="#myModal">
+                Add group
+              </button>
+              <div className="modal fade"
+               id="myModal"
+               role="dialog">
+                <div className="modal-dialog">
+                  <div className="modal-content">
+                    <div className="modal-header">
+                      <button type="button" className="close" data-dismiss="modal">&times;</button>
+                      <h4 className="modal-title">Create Group</h4>
+                    </div>
+                    <div className="modal-body">
+                    <input type="text"
+                     className="form-control"
+                     onChange={this.onChange}
+                      name="groupname"
+                      placeholder="Input groupname...."/>
+                    </div>
+                    <div className="modal-footer">
+                      <button type="button" className="btn btn-danger"
+                       data-dismiss="modal" >Close</button>
+                      <button type="button" className="btn btn-success"
+                       data-dismiss="modal" onClick={ this.onClick} type="submit" name="submit">
+                       Create
+                      </button>
+                    </div>
+                  </div>
+                </div>
+              </div>
+              <div>
+                <ul>
+                <h5>{this.state.groupname === [] ? waiting : this.state.groupname.map((groupName, i) => <li key={i.toString()}>{groupName}</li>)}</h5>
```

Comment:
Too much logic here. The idea is to keep as much logic away from the view. Can we have a function that does this operation here?